### ISSUE TITLE
GEP: HTTPRoute support match by RegularExpression

### ISSUE THREAD

**Issue by rokkiter** [kubernetes-sigs/gateway-api#2694]
**What would you like to be added**:

In [gep-726,](https://github.com/kubernetes-sigs/gateway-api/issues/726). it is mentioned that Regula......dsarExpression will be added to HttpRouter as a Future Extension. Hope to promote the implementation of RegularExpression.

**Why this is needed**:
I think RegularExpression is a function with a very wide range of usage scenarios. If route matching cannot be supported natively, we need to implement it in a very complicated way. For example, in istio, the redirect function cannot support regex matching. I can only use envoyfilter to achieve. Because the Gateway API is a standard, if the Gateway API does not support RegularExpression, the istio community's support for regex matching will also be affected. I believe other communities have similar problems.

Istio issue: https://github.com/istio/istio/issues/47500 for support RegularExpression.

**Comment by arkodg** [comment]
looks like a dup of https://github.com/kubernetes-sigs/gateway-api/issues/2177, @rokkiter you would also need to update the GEP before adding in the API changes

**Comment by rokkiter** [comment]
Yes, our purpose is similar, I am going to create a new gep to drive this

**Comment by youngnick** [comment]
Because doing regex path rewrites involves doing a regex path _match_, followed by using captured info in a rewrite filter, I think that #2177 covers just adding the match criteria as well. See [my comment](https://github.com/kubernetes-sigs/gateway-api/issues/2177#issuecomment-1878144397) on that for more thoughts on what the resulting GEP will need to address.
